### PR TITLE
Bugfix/cmr-7820 - fixing url builder to prevent it from failing on expected requests.

### PR DESCRIPTION
**Pull request by jceaser**
Ticket CMR-7820

It was possible to specify values for the env which caused the url builder function to return a bad CMR URL.

cmr/search/common, cmr/util/common.py and the related test files are the only ones related to the bug fix.

The vast majority of these file changes are pylint requested changes however to bring the code back to zero warnings.

python moved to 3.6 as this is the standard that pylint is working against.

### CODE REVIEW

**Review by mmorahan — APPROVED** [review]

File: CMR/python/cmr/search/common.py [hunk 1]
```diff
@@ -138,6 +138,7 @@ def _cmr_query_url(base: str, query: dict, page_state: dict, config: dict = None
         config: configurations, responds to:
             * env - sit, uat, ops, prod, production, or blank for production
     """
+    #here
```
Comment (jceaser) [inline CMR/python/cmr/search/common.py:141]: remove comment

**Comment by tbs1979** [comment]
looks good

**Review by tbs1979 — APPROVED** [review]

**Comment by jceaser** [comment]
the one comment here I will remove in my next pull request which is just behind this one.

File: CMR/python/cmr/search/common.py [hunk 2]
```diff
@@ -157,18 +158,20 @@ def _cmr_basic_url(base: str, query: dict, config: dict = None):
...
+    base = common.always(base, str)
 
-    config = common.always(config)
-    env = config.get('env', '').lower().strip()
-    if len(env)>0 and not env.endswith("."):
-        env += "."
-    if env in ['', 'ops', 'prod', 'production']:
-        env = ""
+    query = common.always(query)
+    query = '' if len(query)<1 else f'?{net.expand_query_to_parameters(query)}'
```
Comment (charleshuang80) [inline CMR/python/cmr/search/common.py:164]: similar comment to above, nit about spacing around `<` here

File: CMR/python/demos/scripts/vars.py [hunk 3]
```diff
@@ -17,16 +17,16 @@ def get_block_of_records(search, env=None):
     # build a url
     if env is None:
```
Comment (charleshuang80) [inline CMR/python/demos/scripts/vars.py:18]: Just a thought - you are doing this setting the correct url based on env in a few places. Maybe abstract it into a method in common?

**Review by charleshuang80 — APPROVED** [review]

File: CMR/python/cmr/search/common.py (hunk 2 above)
Comment (jceaser) [reply CMR/python/cmr/search/common.py:164]: I will fix this in another ticket, I needed to merge right away. Sorry.

File: CMR/python/cmr/search/common.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jceaser) [reply CMR/python/cmr/search/common.py:141]: yea, I'm fixing this in CMR-7435

**Pull request merged**